### DISCUSSION TITLE
Implement Caching

### DISCUSSION THREAD

**Discussion by Cuica20** [Netflix/dgs-framework#847] [General]
Hi everyone, i am implement a caching in my backend server:

`
public static void main(String[] args) {
		SpringApplication.run(Dsrp2022BackendApplication.class, args);
	}

	@Configuration
	static class PreparsedDocumentProviderConfig {

		private final Cache<String, PreparsedDocumentEntry> cache = Caffeine.newBuilder().maximumSize(250)
				.expireAfterAccess(5, TimeUnit.MINUTES).recordStats().build();

		@Bean
		public PreparsedDocumentProvider preparsedDocumentProvider() {
			return (executionInput, parseAndValidateFunction) -> {
				Function<String, PreparsedDocumentEntry> mapCompute = key -> parseAndValidateFunction.apply(executionInput);
				return cache.get(executionInput.getQuery(), mapCompute);
			};
		}
	}
`

`
@DgsQuery
    public Webinar webinarById(@InputArgument String webinarId) {

        return webinarService.findById(Long.parseLong(webinarId));
    }
`

`
query webinarById($webinarId: ID!) {
   webinarById(webinarId:$webinarId) {
    webinarId
    codeWebinar
    name
    description
    urlPhoto
    webinarSpeaker {
      id
      user {
        userName
        lastName
        biography
        jobTitle
        university
      }
    }
    webinarRegistration {
      id
      user {
        name
        lastName
        email
        biography
      }
    }
  }
}
`

query variables:
`
{
  "webinarId": 1
}
`

My Log:
`
2022-01-28 17:07:36.436  INFO 4612 --- [nio-8081-exec-7] c.d.b.tracing.TracingInstrumentation     : Total execution time: 1020ms
2022-01-28 17:07:36.439  INFO 4612 --- [nio-8081-exec-7] i.StatisticalLoggingSessionEventListener : Session Metrics {
    97323347 nanoseconds spent acquiring 1 JDBC connections;
    0 nanoseconds spent releasing 0 JDBC connections;
    457248 nanoseconds spent preparing 3 JDBC statements;
    399546274 nanoseconds spent executing 3 JDBC statements;
    0 nanoseconds spent executing 0 JDBC batches;
    0 nanoseconds spent performing 0 L2C puts;
    0 nanoseconds spent performing 0 L2C hits;
    0 nanoseconds spent performing 0 L2C misses;
    0 nanoseconds spent executing 0 flushes (flushing a total of 0 entities and 0 collections);
    0 nanoseconds spent executing 0 partial-flushes (flushing a total of 0 entities and 0 collections)
}
`

What am I doing wrong?

**Comment by Cuica20** [comment]
![Captura de Pantalla 2022-01-28 a la(s) 17 11 55](https://user-images.githubusercontent.com/5361567/151628668-3ab06317-451e-4f95-a3b3-fb521f6ae627.png)

**Comment by pushpagarwal** [comment]
PreparsedDocumentProviderConfig only cache Result of Query Parsing step. Query would still be executed as normal flow. It doesn't cache response of Query Execution. 
What you are trying of cache: Result of Query Parsing step or Query responses?

**Reply by Cuica20** [reply]
query response @pushpagarwal

**Reply by setchy** [reply]
Some information in [this discussions thread](https://github.com/Netflix/dgs-framework/discussions/821#discussioncomment-1956973) about `@cacheControl` schema directive being added to `federation-jvm:0.8.0`.  

🤞 should be coming to dgs-framework shortly
